src/main/java/hudson/scm/SubversionSCM.java

```diff
@@ -1392,7 +1392,13 @@ else if (project.getLastBuild()!=null) {
 
         final Map<String,ISVNAuthenticationProvider> authProviders = new LinkedHashMap<String,
                 ISVNAuthenticationProvider>();
-        for (ModuleLocation loc: getLocations()) {
+
+        EnvVars env = null;
+        if (project.getLastCompletedBuild() != null) {
```

Comment:
Wrong construct, which will make the static analysis angry (e.g. cleanup of builds between two calls). You should assign the last completed build to a variable and then check&use it